### PR TITLE
Improve load performance

### PR DESCRIPTION
**Pull request by yah01**
Signed-off-by: yah01 <yang.cen@zilliz.com>
#15736 #13264
/kind improvement

### CODE REVIEW

**Comment by yah01** [comment]
rerun ut

**Comment by yah01** [comment]
rerun ut

**Comment by yah01** [comment]
rerun ut

File: internal/util/funcutil/parallel.go [hunk 1]
```diff
@@ -31,13 +32,22 @@ func GetFunctionName(i interface{}) string {
...
+type DataProcessFunc func(data interface{}) error
+
+// ProcessFuncParallel processes function in parallel.
 //
 // ProcessFuncParallel waits for all goroutines done if no errors occur.
 // If some goroutines return error, ProcessFuncParallel cancels other goroutines as soon as possible and wait
 // for all other goroutines done, and returns the first error occurs.
 // Reference: https://stackoverflow.com/questions/40809504/idiomatic-goroutine-termination-and-error-handling
-func ProcessFuncParallel(total, maxParallel int, f func(idx int) error, fname string) error {
+func ProcessFuncParallel(total, maxParallel int, f ProcessFunc, fname string) error {
```
Comment (xiaofan-luan) [inline internal/util/funcutil/parallel.go:45]: https://github.com/git-hulk/routines
https://github.com/goinggo/workpool/blob/master/workpool.go
recommend to use some routine pool(not sure the one listed is the best choise), which could be more elegant and

File: internal/util/funcutil/parallel.go [hunk 2]
```diff
@@ -124,3 +254,101 @@ func ProcessFuncParallel(total, maxParallel int, f func(idx int) error, fname st
...
+
+				select {
+				case <-quit:
+					return
+				default:
+				}
+
+				err := f(idx)
+				if err != nil {
+					log.Debug(fname, zap.Error(err), zap.Any("idx", idx))
```
Comment (xiaofan-luan) [inline internal/util/funcutil/parallel.go:301]: error log need to be in log.Warn or error

**Comment by yah01** [comment]
Close this PR, lots of codes should be modified due to conflicts and OOM problem. I will create a new one and reference this later.

**Pull request closed without merging**